server/queue/queue-api/src/test/java/org/apache/james/queue/api/PriorityManageableMailQueueContract.java

```diff
@@ -0,0 +1,113 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.api;
+
+import static org.apache.james.queue.api.MailQueueFixture.NAME0;
+import static org.apache.james.queue.api.MailQueueFixture.NAME1;
+import static org.apache.james.queue.api.MailQueueFixture.NAME2;
+import static org.apache.james.queue.api.MailQueueFixture.NAME3;
+import static org.apache.james.queue.api.MailQueueFixture.NAME4;
+import static org.apache.james.queue.api.MailQueueFixture.NAME5;
+import static org.apache.james.queue.api.MailQueueFixture.NAME6;
+import static org.apache.james.queue.api.MailQueueFixture.NAME7;
+import static org.apache.james.queue.api.MailQueueFixture.NAME8;
+import static org.apache.james.queue.api.MailQueueFixture.NAME9;
+import static org.apache.james.queue.api.MailQueueFixture.createMimeMessage;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Date;
+
+import javax.mail.MessagingException;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.jupiter.api.Test;
+
+public interface PriorityManageableMailQueueContract {
+
+    ManageableMailQueue getManageableMailQueue();
+
+    @Test
+    default void browseShouldBeOrderedByPriority() throws Exception {
```

Comment:
why do you need to duplicate this test from the other contrat ? you can very well extends previous contract instead

Reply:
true.

Reply:
But this one is ordered by priority and is unique. Maybe there is a misunderstood here, but this test is unique, do not overload something else .